### PR TITLE
Feature/use sample tracking v2

### PR DESCRIPTION
**Pull request by wow-such-code**
use new location-independent sample tracker to set "data available" status.

### CODE REVIEW

File: drop-boxes/register-mtb-data-dropbox/register-mtb-data-dropbox.py [hunk 1]
```diff
@@ -128,10 +128,10 @@
 SERVICE_CREDENTIALS.user = tracking_helper.get_service_user()
 SERVICE_CREDENTIALS.password = tracking_helper.get_service_password()
 SERVICE_REGISTRY_URL = URL(tracking_helper.get_service_reg_url())
-QBIC_LOCATION = tracking_helper.get_qbic_location_json()
+DATA_AVAILABLE_JSON = tracking_helper.get_data_available_status_json()
 
-### We need this object to update the sample location later
-SAMPLE_TRACKER = SampleTracker.createQBiCSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, QBIC_LOCATION)
+### We need this object to update the sample status later
+SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, DATA_AVAILABLE_JSON)
```
Comment (jenniferboedker) [inline drop-boxes/register-mtb-data-dropbox/register-mtb-data-dropbox.py]: why do we need to set this in one file two times? You basically overwrite the same variables with the same values

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-projects-dropbox.py [hunk 2]
```diff
@@ -126,10 +126,10 @@
 SERVICE_CREDENTIALS.user = tracking_helper.get_service_user()
 SERVICE_CREDENTIALS.password = tracking_helper.get_service_password()
 SERVICE_REGISTRY_URL = URL(tracking_helper.get_service_reg_url())
-QBIC_LOCATION = tracking_helper.get_qbic_location_json()
+DATA_AVAILABLE_JSON = tracking_helper.get_data_available_status_json()
 
-### We need this object to update the sample location later
-SAMPLE_TRACKER = SampleTracker.createQBiCSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, QBIC_LOCATION)
+### We need this object to update the sample status later
+SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, DATA_AVAILABLE_JSON)
```
Comment (jenniferboedker) [inline drop-boxes/register-mtb-projects-dropbox/register-mtb-projects-dropbox.py]: here as well

File: drop-boxes/register-qpcr-dropbox/register-qpcr-dropbox.py [hunk 3]
```diff
@@ -26,10 +26,10 @@
 SERVICE_CREDENTIALS.user = tracking_helper.get_service_user()
 SERVICE_CREDENTIALS.password = tracking_helper.get_service_password()
 SERVICE_REGISTRY_URL = URL(tracking_helper.get_service_reg_url())
-QBIC_LOCATION = tracking_helper.get_qbic_location_json()
+DATA_AVAILABLE_JSON = tracking_helper.get_data_available_status_json()
 
-### We need this object to update the sample location later
-SAMPLE_TRACKER = SampleTracker.createQBiCSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, QBIC_LOCATION)
+### We need this object to update the sample status later
+SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, DATA_AVAILABLE_JSON)
```
Comment (jenniferboedker) [inline drop-boxes/register-qpcr-dropbox/register-qpcr-dropbox.py]: In general: why do we need to provide the single status here for setting up the sample tracker?

File: drop-boxes/register-wf-16staxonomicprofiling/register-16staxonomic-dropbox.py [hunk 4]
```diff
@@ -28,10 +28,10 @@
 SERVICE_CREDENTIALS.user = tracking_helper.get_service_user()
 SERVICE_CREDENTIALS.password = tracking_helper.get_service_password()
 SERVICE_REGISTRY_URL = URL(tracking_helper.get_service_reg_url())
-QBIC_LOCATION = tracking_helper.get_qbic_location_json()
+DATA_AVAILABLE_JSON = tracking_helper.get_data_available_status_json()
 
-### We need this object to update the sample location later
-SAMPLE_TRACKER = SampleTracker.createQBiCSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, QBIC_LOCATION)
+### We need this object to update the sample status later
+SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, DATA_AVAILABLE_JSON)
```
Comment (jenniferboedker) [inline drop-boxes/register-wf-16staxonomicprofiling/register-16staxonomic-dropbox.py]: not used?

File: drop-boxes/test-dropbox/etl_msconvert.py [hunk 5]
```diff
@@ -484,4 +484,4 @@ def process(transaction):
     transaction.moveFile(incomingPath, dataSet)
 
     # Update Sample Status
-    sampleTracker.updateSampleLocationToCurrentLocation(code)
+    sampleTracker.updateSampleStatus(code)
```
Comment (jenniferboedker) [inline drop-boxes/test-dropbox/etl_msconvert.py]: here you forgot to update the variables like in the files above?

File: drop-boxes/test-dropbox/script.py [hunk 6]
```diff
@@ -77,7 +77,7 @@ def process(transaction):
         print qbicLocation
 
         # Update Sample Location
-        sampleTracker.updateSampleLocationToCurrentLocation(sampleID)
+        sampleTracker.updateSampleStatus(sampleID)
```
Comment (jenniferboedker) [inline drop-boxes/test-dropbox/script.py]: here also? Update the variables and sample tracker to be able to call the new function?

**Review by jenniferboedker — CHANGES_REQUESTED** [review]
Left some questions to be discussed :)

File: drop-boxes/register-wf-16staxonomicprofiling/register-16staxonomic-dropbox.py (hunk 4 above)
Comment (wow-such-code) [reply drop-boxes/register-wf-16staxonomicprofiling/register-16staxonomic-dropbox.py]: good catch, removed

File: drop-boxes/test-dropbox/etl_msconvert.py (hunk 5 above)
Comment (wow-such-code) [reply drop-boxes/test-dropbox/etl_msconvert.py]: not sure what this script did here, was probably a test, removed

File: drop-boxes/test-dropbox/script.py (hunk 6 above)
Comment (wow-such-code) [reply drop-boxes/test-dropbox/script.py]: good catch, has been added now

File: drop-boxes/register-qpcr-dropbox/register-qpcr-dropbox.py (hunk 3 above)
Comment (wow-such-code) [reply drop-boxes/register-qpcr-dropbox/register-qpcr-dropbox.py]: good point, this is because it was implemented quite similar to the qbic sample tracker with its location...would need to change it in the lib, then...should I?

File: drop-boxes/register-mtb-data-dropbox/register-mtb-data-dropbox.py (hunk 1 above)
Comment (wow-such-code) [reply drop-boxes/register-mtb-data-dropbox/register-mtb-data-dropbox.py]: duplicate has been removed, thanks

File: drop-boxes/register-iontorrent-data/register-iontorrent-data.py [hunk 7]
```diff
@@ -39,7 +39,7 @@
 DATA_AVAILABLE_JSON = tracking_helper.get_data_available_status_json()
 
 ### We need this object to update the sample status later
-SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, DATA_AVAILABLE_JSON)
+SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS)
```
Comment (jenniferboedker) [inline drop-boxes/register-iontorrent-data/register-iontorrent-data.py:42]: not used?

File: drop-boxes/register-mtb-data-dropbox/register-mtb-data-dropbox.py [hunk 8]
```diff
@@ -83,7 +83,7 @@
 DATA_AVAILABLE_JSON = tracking_helper.get_data_available_status_json()
 
 ### We need this object to update the sample status later
-SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, DATA_AVAILABLE_JSON)
+SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS)
```
Comment (jenniferboedker) [inline drop-boxes/register-mtb-data-dropbox/register-mtb-data-dropbox.py:86]: not used? defined in 134 again

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-projects-dropbox.py [hunk 9]
```diff
@@ -121,17 +119,6 @@
...
-SERVICE_CREDENTIALS = ServiceCredentials()
-SERVICE_CREDENTIALS.user = tracking_helper.get_service_user()
-SERVICE_CREDENTIALS.password = tracking_helper.get_service_password()
-SERVICE_REGISTRY_URL = URL(tracking_helper.get_service_reg_url())
-DATA_AVAILABLE_JSON = tracking_helper.get_data_available_status_json()
-
-### We need this object to update the sample status later
-SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, DATA_AVAILABLE_JSON)
-
-
```
Comment (jenniferboedker) [inline drop-boxes/register-mtb-projects-dropbox/register-mtb-projects-dropbox.py]: 👍🏼

File: drop-boxes/test-dropbox/etl_msconvert.py [hunk 10]
```diff
@@ -1,487 +0,0 @@
-"""
-This etl script converts proteomics raw files to mzML.
-
-Incoming raw files are converted and the new mzML is copied
```
Comment (jenniferboedker) [inline drop-boxes/test-dropbox/etl_msconvert.py:4]: why did you remove the file?

Comment (wow-such-code) [reply drop-boxes/test-dropbox/etl_msconvert.py:4]: it's not used and the dropbox in question is for testing anyway

File: drop-boxes/register-mtb-data-dropbox/register-mtb-data-dropbox.py (hunk 8 above)
Comment (wow-such-code) [reply drop-boxes/register-mtb-data-dropbox/register-mtb-data-dropbox.py:86]: it's used, but I will remove the duplicate definition

**Review by jenniferboedker — APPROVED** [review]
